src/request_callback.cpp

```diff
@@ -91,6 +99,23 @@ int32_t RequestCallback::encode(int version, int flags, BufferVec* bufs) {
   return length + header_size;
 }
 
+ICompressor::Buffer RequestCallback::compress(const BufferVec& bufs, ICompressor* compressor) const {
```

Comment:
I am not sure if this method is needed; operate off of the original `BufferVec` and allow for the new compressed `BufferVec` to be returned from the compression algorithm along with a flag to determine if the compression was successful.